po-docs/ja/virt-edit.1
.\" Automatically generated by Podwrapper::Man 1.40.2 (Pod::Simple 3.35)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is >0, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{\
.    if \nF \{\
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{\
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\" ========================================================================
.\"
.IX Title "virt-edit 1"
.TH virt-edit 1 "2019-02-07" "libguestfs-1.40.2" "Virtualization Support"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "名前"
.IX Header "名前"
virt-edit \- 仮想マシンのファイルの編集
.SH "書式"
.IX Header "書式"
.Vb 1
\& virt\-edit [\-\-options] \-d domname file [file ...]
\&
\& virt\-edit [\-\-options] \-a disk.img [\-a disk.img ...] file [file ...]
\&
\& virt\-edit [\-d domname|\-a disk.img] file \-e \*(Aqexpr\*(Aq
.Ve
.PP
古い形式:
.PP
.Vb 1
\& virt\-edit domname file
\&
\& virt\-edit disk.img [disk.img ...] file
.Ve
.SH "説明"
.IX Header "説明"
\&\f(CW\*(C`virt\-edit\*(C'\fR は名前付きの仮想マシン（またはディスクイメージ）にある、それぞれの \f(CW\*(C`ファイル\*(C'\fR を編集するためのコマンドラインツールです。
.PP
複数のファイル名を指定できます。その場合、それぞれ順番に編集されます。各ファイル名はルートディレクトリーから始まる（つまり '/'
から始まる）完全パスである必要があります。
.PP
単にファイルを表示したいだけならば、 \fBvirt\-cat\fR\|(1) を使用してください。
.PP
より複雑な場合に対しては \fBguestfish\fR\|(1) ツールを参照してください（以下の \*(L"guestfish の使用法\*(R" 参照）。
.PP
\&\f(CW\*(C`virt\-edit\*(C'\fR は新しいファイルを作成するために使用できません。 \fBguestfish\fR\|(1) はこのことやさらにいろいろなことができます。
.SH "例"
.IX Header "例"
名前付きのファイルを対話的に編集します:
.PP
.Vb 1
\& virt\-edit \-d mydomain /boot/grub/grub.conf
\&
\& virt\-edit \-d mydomain /etc/passwd
.Ve
.PP
Windows 仮想マシンに対しては、Windows のパスが解釈されます:
.PP
.Vb 1
\& virt\-edit \-d mywindomain \*(Aqc:\eautoexec.bat\*(Aq
.Ve
.PP
Perl がインストールされていると、非対話的にファイルを編集することもできます（以下の \*(L"非対話的な編集方法\*(R" 参照）。 init
の初期ランベルを 5 に変更するには:
.PP
.Vb 1
\& virt\-edit \-d mydomain /etc/inittab \-e \*(Aqs/^id:.*/id:5:initdefault:/\*(Aq
.Ve
.SH "オプション"
.IX Header "オプション"
.IP "\fB\-\-help\fR" 4
.IX Item "--help"
簡単なヘルプを表示します。
.IP "\fB\-a\fR file" 4
.IX Item "-a file"
.PD 0
.IP "\fB\-\-add\fR file" 4
.IX Item "--add file"
.PD
Add \fIfile\fR which should be a disk image from a virtual machine.  If the
virtual machine has multiple block devices, you must supply all of them with
separate \fI\-a\fR options.
.Sp
ディスクイメージの形式は自動検知されます。 これを上書きして強制的に特定の形式を使用する場合、 \fI\-\-format=..\fR オプションを使用します。
.IP "\fB\-a\fR \s-1URI\s0" 4
.IX Item "-a URI"
.PD 0
.IP "\fB\-\-add\fR \s-1URI\s0" 4
.IX Item "--add URI"
.PD
リモートディスクを追加します。 \*(L"リモートストレージの追加\*(R" in \fBguestfish\fR\|(1) 参照。
.IP "\fB\-b\fR \s-1EXTENSION\s0" 4
.IX Item "-b EXTENSION"
.PD 0
.IP "\fB\-\-backup\fR \s-1EXTENSION\s0" 4
.IX Item "--backup EXTENSION"
.PD
\&\fI仮想マシンのディスクイメージにある\fR 元のファイルのバックアップを作成します。バックアップは元のファイル名に \f(CW\*(C`extension\*(C'\fR
を加えたものになります。
.Sp
一般的に \f(CW\*(C`extension\*(C'\fR の最初の文字はドット \f(CW\*(C`.\*(C'\fR です。そのため次のように書きます:
.Sp
.Vb 1
\& virt\-edit \-b .orig [etc]
.Ve
.Sp
初期設定では、バックアップファイルは作成されません。
.IP "\fB\-c\fR \s-1URI\s0" 4
.IX Item "-c URI"
.PD 0
.IP "\fB\-\-connect\fR \s-1URI\s0" 4
.IX Item "--connect URI"
.PD
libvirt を使用していると、指定された \fI\s-1URI\s0\fR に接続します。  省略すると、デフォルトの libvirt ハイパーバイザーに接続します。
.Sp
仮想マシンのブロックデバイスを直接指定すると、libvirt はまったく使用されません。
.IP "\fB\-d\fR \s-1GUEST\s0" 4
.IX Item "-d GUEST"
.PD 0
.IP "\fB\-\-domain\fR \s-1GUEST\s0" 4
.IX Item "--domain GUEST"
.PD
名前付きの libvirt 仮想マシンからすべてのディスクを追加します。  名前の代わりに仮想マシンの \s-1UUID\s0 を使用できます。
.IP "\fB\-\-echo\-keys\fR" 4
.IX Item "--echo-keys"
キーやパスフレーズを入力するとき、通常 virt-edit はエコーを無効化します。 そのため、入力内容を確認できません。 テンペスト攻撃の心配がなく、
部屋に誰も居なければ、 入力内容を確認するためにこのフラグを指定できます。
.IP "\fB\-e\fR \s-1EXPR\s0" 4
.IX Item "-e EXPR"
.PD 0
.IP "\fB\-\-edit\fR \s-1EXPR\s0" 4
.IX Item "--edit EXPR"
.IP "\fB\-\-expr\fR \s-1EXPR\s0" 4
.IX Item "--expr EXPR"
.PD
外部エディターを起動する代わりに、ファイルにある各行に対して非対話的に Perl 表現 \f(CW\*(C`EXPR\*(C'\fR を適用します。以下の \*(L"非対話的な編集方法\*(R"
を参照してください。
.Sp
表現がシェルにより変更されるのを防ぐために、適切に引用符でくくるよう注意してください。
.Sp
このオプションは Perl 5 がインストールされているときのみ利用可能であることに注意してください。
.IP "\fB\-\-format=raw|qcow2|..\fR" 4
.IX Item "--format=raw|qcow2|.."
.PD 0
.IP "\fB\-\-format\fR" 4
.IX Item "--format"
.PD
\&\fI\-a\fR オプションは標準状態でディスクイメージの形式を自動検知します。 これを使用することにより、コマンドラインで後続の \fI\-a\fR
オプションのディスク形式を強制的に指定できます。 引数なしで \fI\-\-format\fR を使用することにより、 後続の \fI\-a\fR
オプションに対して自動検知に戻せます。
.Sp
例:
.Sp
.Vb 1
\& virt\-edit \-\-format=raw \-a disk.img file
.Ve
.Sp
forces raw format (no auto-detection) for \fIdisk.img\fR.
.Sp
.Vb 1
\& virt\-edit \-\-format=raw \-a disk.img \-\-format \-a another.img file
.Ve
.Sp
forces raw format (no auto-detection) for \fIdisk.img\fR and reverts to
auto-detection for \fIanother.img\fR.
.Sp
仮想マシンのディスクイメージが信頼できない raw 形式である場合、 ディスク形式を指定するためにこのオプションを使用すべきです。
これにより、悪意のある仮想マシンにより起こり得る セキュリティ問題を回避できます (\s-1CVE\-2010\-3851\s0)。
.IP "\fB\-\-key\fR \s-1SELECTOR\s0" 4
.IX Item "--key SELECTOR"
Specify a key for \s-1LUKS,\s0 to automatically open a \s-1LUKS\s0 device when using the
inspection.  \f(CW\*(C`SELECTOR\*(C'\fR can be in one of the following formats:
.RS 4
.ie n .IP "\fB\-\-key\fR ""DEVICE"":key:KEY_STRING" 4
.el .IP "\fB\-\-key\fR \f(CWDEVICE\fR:key:KEY_STRING" 4
.IX Item "--key DEVICE:key:KEY_STRING"
Use the specified \f(CW\*(C`KEY_STRING\*(C'\fR as passphrase.
.ie n .IP "\fB\-\-key\fR ""DEVICE"":file:FILENAME" 4
.el .IP "\fB\-\-key\fR \f(CWDEVICE\fR:file:FILENAME" 4
.IX Item "--key DEVICE:file:FILENAME"
Read the passphrase from \fI\s-1FILENAME\s0\fR.
.RE
.RS 4
.RE
.IP "\fB\-\-keys\-from\-stdin\fR" 4
.IX Item "--keys-from-stdin"
Read key or passphrase parameters from stdin.  The default is to try to read
passphrases from the user by opening \fI/dev/tty\fR.
.IP "\fB\-m\fR dev[:mountpoint[:options[:fstype]]]" 4
.IX Item "-m dev[:mountpoint[:options[:fstype]]]"
.PD 0
.IP "\fB\-\-mount\fR dev[:mountpoint[:options[:fstype]]]" 4
.IX Item "--mount dev[:mountpoint[:options[:fstype]]]"
.PD
指定されたマウントポイントにある、 名前付きパーティションまたは論理ボリュームをマウントします。
.Sp
If the mountpoint is omitted, it defaults to \fI/\fR.
.Sp
Specifying any mountpoint disables the inspection of the guest and the mount
of its root and all of its mountpoints, so make sure to mount all the
mountpoints needed to work with the filenames given as arguments.
.Sp
If you don’t know what filesystems a disk image contains, you can either run
guestfish without this option, then list the partitions, filesystems and LVs
available (see \*(L"list-partitions\*(R", \*(L"list-filesystems\*(R" and \*(L"lvs\*(R"
commands), or you can use the \fBvirt\-filesystems\fR\|(1) program.
.Sp
マウントパラメーターの三番目の (ほとんど使用されない) 項目は、 バックエンドのファイルシステムをマウントするために使用される
マウントオプションの一覧です。 これが指定されていない場合、 マウントオプションは空文字列または \f(CW\*(C`ro\*(C'\fR (\fI\-\-ro\fR
フラグが使用されている場合) になります。
.Sp
.Vb 1
\& \-m /dev/sda1:/:acl,user_xattr
.Ve
.Sp
このフラグを使用することは、 \f(CW\*(C`mount\-options\*(C'\fR コマンドを使用することと同等です。
.Sp
パラメーターの四番目の項目は使用するファイルシステムドライバー (\f(CW\*(C`ext3\*(C'\fR や \f(CW\*(C`ntfs\*(C'\fR など) です。 これはほとんど必要ありません。
しかし、複数のドライバーがファイルシステムに対して有効である場合 (例: \f(CW\*(C`ext2\*(C'\fR と \f(CW\*(C`ext3\*(C'\fR)、 または libguestfs
がファイルシステムを誤検知している場合、 これは有用です。
.IP "\fB\-v\fR" 4
.IX Item "-v"
.PD 0
.IP "\fB\-\-verbose\fR" 4
.IX Item "--verbose"
.PD
デバッグ用の冗長なメッセージを有効にします。
.IP "\fB\-V\fR" 4
.IX Item "-V"
.PD 0
.IP "\fB\-\-version\fR" 4
.IX Item "--version"
.PD
バージョン番号を表示して、終了します。
.IP "\fB\-x\fR" 4
.IX Item "-x"
libguestfs \s-1API\s0 呼び出しのトレースを有効にします。
.SH "旧形式のコマンドライン引数"
.IX Header "旧形式のコマンドライン引数"
前のバージョンの virt-edit はどちらを書くこともできます:
.PP
.Vb 1
\& virt\-edit disk.img [disk.img ...] file
.Ve
.PP
または
.PP
.Vb 1
\& virt\-edit guestname file
.Ve
.PP
whereas in this version you should use \fI\-a\fR or \fI\-d\fR respectively to avoid
the confusing case where a disk image might have the same name as a guest.
.PP
互換性のため古い形式がまだサポートされています。
.SH "非対話的な編集方法"
.IX Header "非対話的な編集方法"
\&\f(CW\*(C`virt\-edit\*(C'\fR normally calls out to \f(CW$EDITOR\fR (or vi) so the system
administrator can interactively edit the file.
.PP
There are two ways also to use \f(CW\*(C`virt\-edit\*(C'\fR from scripts in order to make
automated edits to files.  (Note that although you \fIcan\fR use \f(CW\*(C`virt\-edit\*(C'\fR
like this, it’s less error-prone to write scripts directly using the
libguestfs \s-1API\s0 and Augeas for configuration file editing.)
.PP
The first method is to temporarily set \f(CW$EDITOR\fR to any script or program
you want to run.  The script is invoked as \f(CW\*(C`$EDITOR tmpfile\*(C'\fR and it should
update \f(CW\*(C`tmpfile\*(C'\fR in place however it likes.
.PP
The second method is to use the \fI\-e\fR parameter of \f(CW\*(C`virt\-edit\*(C'\fR to run a
short Perl snippet in the style of \fBsed\fR\|(1).  For example to replace all
instances of \f(CW\*(C`foo\*(C'\fR with \f(CW\*(C`bar\*(C'\fR in a file:
.PP
.Vb 1
\& virt\-edit \-d domname filename \-e \*(Aqs/foo/bar/\*(Aq
.Ve
.PP
The full power of Perl regular expressions can be used (see \fBperlre\fR\|(1)).
For example to delete root’s password you could do:
.PP
.Vb 1
\& virt\-edit \-d domname /etc/passwd \-e \*(Aqs/^root:.*?:/root::/\*(Aq
.Ve
.PP
What really happens is that the snippet is evaluated as a Perl expression
for each line of the file.  The line, including the final \f(CW\*(C`\en\*(C'\fR, is passed
in \f(CW$_\fR and the expression should update \f(CW$_\fR or leave it unchanged.
.PP
To delete a line, set \f(CW$_\fR to the empty string.  For example, to delete the
\&\f(CW\*(C`apache\*(C'\fR user account from the password file you can do:
.PP
.Vb 1
\& virt\-edit \-d mydomain /etc/passwd \-e \*(Aq$_ = "" if /^apache:/\*(Aq
.Ve
.PP
To insert a line, prepend or append it to \f(CW$_\fR.  However appending lines to
the end of the file is rather difficult this way since there is no concept
of \*(L"last line of the file\*(R" \- your expression just doesn't get called again.
You might want to use the first method (setting \f(CW$EDITOR\fR) if you want to
do this.
.PP
The variable \f(CW$lineno\fR contains the current line number.  As is
traditional, the first line in the file is number \f(CW1\fR.
.PP
The return value from the expression is ignored, but the expression may call
\&\f(CW\*(C`die\*(C'\fR in order to abort the whole program, leaving the original file
untouched.
.PP
Remember when matching the end of a line that \f(CW$_\fR may contain the final
\&\f(CW\*(C`\en\*(C'\fR, or (for \s-1DOS\s0 files) \f(CW\*(C`\er\en\*(C'\fR, or if the file does not end with a
newline then neither of these.  Thus to match or substitute some text at the
end of a line, use this regular expression:
.PP
.Vb 1
\& /some text(\er?\en)?$/
.Ve
.PP
Alternately, use the perl \f(CW\*(C`chomp\*(C'\fR function, being careful not to chomp
\&\f(CW$_\fR itself (since that would remove all newlines from the file):
.PP
.Vb 1
\& my $m = $_; chomp $m; $m =~ /some text$/
.Ve
.SH "Windows のパス"
.IX Header "Windows のパス"
\&\f(CW\*(C`virt\-edit\*(C'\fR has a limited ability to understand Windows drive letters and
paths (eg. \fIE:\efoo\ebar.txt\fR).
.PP
仮想マシンが Windows を実行している場合に限り:
.IP "\(bu" 4
Drive letter prefixes like \f(CW\*(C`C:\*(C'\fR are resolved against the Windows Registry
to the correct filesystem.
.IP "\(bu" 4
Any backslash (\f(CW\*(C`\e\*(C'\fR) characters in the path are replaced with forward
slashes so that libguestfs can process it.
.IP "\(bu" 4
The path is resolved case insensitively to locate the file that should be
edited.
.PP
There are some known shortcomings:
.IP "\(bu" 4
Some \s-1NTFS\s0 symbolic links may not be followed correctly.
.IP "\(bu" 4
\&\s-1NTFS\s0 junction points that cross filesystems are not followed.
.SH "guestfish の使用法"
.IX Header "guestfish の使用法"
\&\fBguestfish\fR\|(1) is a more powerful, lower level tool which you can use when
\&\f(CW\*(C`virt\-edit\*(C'\fR doesn't work.
.PP
Using \f(CW\*(C`virt\-edit\*(C'\fR is approximately equivalent to doing:
.PP
.Vb 1
\& guestfish \-\-rw \-i \-d domname edit /file
.Ve
.PP
where \f(CW\*(C`domname\*(C'\fR is the name of the libvirt guest, and \fI/file\fR is the full
path to the file.
.PP
The command above uses libguestfs’s guest inspection feature and so does not
work on guests that libguestfs cannot inspect, or on things like arbitrary
disk images that don't contain guests.  To edit a file on a disk image
directly, use:
.PP
.Vb 1

ChangeLog
commit 91698a8500d19e06811133bfab27d1e1e65a583e
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Thu Feb 7 14:45:14 2019 +0000

    Version 1.40.2.

commit c7b374a757821bbd8f369fa268a75485163b52d7
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Wed Feb 6 15:19:53 2019 +0100

    inspect: revamp icon extraction for *SUSE guests
    
    Newer versions do not have the 24px distributor.png icon; OTOH they have
    the 48px version, so look for that one before the 24px one.
    
    Also, bump the size limit to 10K, as newer versions of the icon are
    bigger than 3K.
    
    (cherry picked from commit f41a75cea67dc3f2e18700303fac5d7078b3291a)

commit 8e62f30293737472ec6b9a2a238bb1206b2fbf4e
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Wed Feb 6 15:15:08 2019 +0100

    inspect: factor out find_png helper function
    
    Isolate the code for looking for a PNG file among a specified list from
    the icon_ubuntu implementation, and switch that function to it.
    
    This is just code refactoring, with no behaviour changes.
    
    (cherry picked from commit 6ff95acc72b02c4df0520e8cae7190447e4b2da5)

commit 178d9d821af6ee25f9f5141c20e4de42315de25f
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Wed Feb 6 15:05:51 2019 +0100

    inspect: bump size limit for Mageia guests
    
    Newer versions of the logo in Cauldron are almost 3K, so bump the limit
    to 10K for now.
    
    (cherry picked from commit 032b8d4c72ab233922482871403c2d9fe2f4a75b)

commit cc83aae48dbda40afa1b1652bd1a53dc9dd61573
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Tue Jan 29 15:07:18 2019 +0100

    v2v: linux: use NEVR for querying RPM packages (RHBZ#1669395)
    
    Use NEVR when querying RPM for the list of files of a package, instead
    of ENVR.  Also, use the epoch only when non-zero, and version of RPM
    supports it.
    
    The approach is basically copied from what supermin does in its RPM
    package handler.
    
    (cherry picked from commit 5eafd1273132b79018ddb72426b128f0fb8a39f4)

commit 76797064449a99351ca44586b16a50c5b74c0a47
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Mon Jan 21 15:59:45 2019 +0000

    build: Reduce -Wformat-overflow=2 to =1.
    
    GCC 9 gives this error:
    
    qemuopts.c: In function 'qemuopts_to_config_channel':
    qemuopts.c:987:29: error: '%.*s' directive output between 0 and 2147483647 bytes may exceed minimum required size of 4095 [-Werror=format-overflow=]
      987 |             fprintf (fp, "  %.*s = ", (int) k, values[j]);
          |                             ^~~~
    qemuopts.c:987:26: note: assuming directive output of 1 byte
      987 |             fprintf (fp, "  %.*s = ", (int) k, values[j]);
          |                          ^~~~~~~~~~~
    
    See discussion at the following GCC bug:
    
    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88993
    (cherry picked from commit b26bd778b58b78eb59503413fd5b41ac23725690)

commit d5dc8fba74edb39bede2e42316f3bd0e9c26c24c
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Tue Jan 22 13:04:59 2019 +0100

    python: change types for RBufferOut/FBuffer with Python 3 (RHBZ#1661871)
    
    So far RBufferOut return values, and FBuffer struct fields are 'str' on
    all the versions of Python.  Python 3 distinguishes between 'str'
    (unicode strings), and 'bytes', with 'str' no more able to hold
    arbitrary data.
    
    For this reason, switch the return value of RBufferOut functions, and
    FBuffer struct fields to bytes on Python 3: while this is a potentially
    incompatibile change, this is the only way to handle safely sequences
    of arbitrary bytes.
    
    (cherry picked from commit 0ee02e0117527b86a31b2a88a14994ce7f15571f)

commit c18957e0ea4233e2237e68c7d5e647cbde69e764
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Tue Jan 22 12:43:11 2019 +0100

    python: fix call of Python handlers of events
    
    Make sure to reference the arguments, to make sure they are kept alive
    during the function call; this is visible when setting an event handler
    for the CLOSE event, and testing it with Python 3.
    
    This does not seem to create a memory leak e.g. with Python 2.
    
    Also, switch away from the quasi-internal PyEval_CallObject to the
    public PyObject_CallObject, which takes care of doing safety checks.
    
    (cherry picked from commit 85235aec837716f1ddb2926b9a59a02543195500)

commit 53a7f2ffec1f7b014db4909371e0c1b1576ba10a
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Fri Jan 18 13:22:26 2019 +0100

    build: integrate ocaml-link.sh with automake silent rules
    
    In case either the silent rules are disabled, or V=1 is set, print the
    full command line that is executed.
    
    (cherry picked from commit a8006afaf545d5188f592d8583f9dc373ded7671)

commit 1f026244ba5891a00617f4ab8c41b8680ff8c497
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Thu Jan 17 12:06:36 2019 +0000

    Version 1.40.1.

commit ae2aa14527605ffb89d86bf93bf8ca4f969118e1
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Thu Jan 17 12:03:07 2019 +0000

    valgrind: Add workaround for slow loading of debuginfo by valgrind.
    
    See nbdkit commit
    https://github.com/libguestfs/nbdkit/commit/bd32d899edb4c592a2df0ce4b26a2cf8396eba1d
    and https://bugzilla.redhat.com/show_bug.cgi?id=1662656
    
    (cherry picked from commit ea7c13fed7922b233db8382b1348dfb436e3ba85)

commit 1de72f8f00600bcaa76742dcfd366fb702eef3c2
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Tue Jan 15 14:39:41 2019 +0000

    Version 1.40.0.

commit 6151d8694d9db0d709a9ad807a5ca6a098d6502c
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Wed Jan 16 11:49:45 2019 +0000

    po: Drop ca and pt_BR from linguas.
    
    Commit 4252a490bb117940c1d38c838345ed110457386e dropped the PO files
    for ca (Catalan) and pt_BR (Brazilian Portuguese).  However it did not
    drop these languages from the linguas line in the Makefile resulting

po-docs/uk/virt-edit.pl

=head1 НАЗВА

virt-edit - Edit a file in a virtual machine

=head1 КОРОТКИЙ ОПИС

 virt-edit [--options] domname file

 virt-edit [--options] disk.img [disk.img ...] file

 virt-edit [domname|disk.img] file -e 'expr'

=head1 WARNING

You must I<not> use C<virt-edit> on live virtual machines.  If you do this,
you risk disk corruption in the VM.  C<virt-edit> tries to stop you from
doing this, but doesn't catch all cases.

=head1 ОПИС

C<virt-edit> is a command line tool to edit C<file> where C<file> exists in
the named virtual machine (or disk image).

If you want to just view a file, use L<virt-cat(1)>.

For more complex cases you should look at the L<guestfish(1)> tool (see
L</USING GUESTFISH> below).

C<virt-edit> cannot be used to create a new file, nor to edit multiple
files.  L<guestfish(1)> can do that and much more.

=head1 EXAMPLES

Edit the named files interactively:

 virt-edit mydomain /boot/grub/grub.conf

 virt-edit mydomain /etc/passwd

For Windows guests, some Windows paths are understood:

 virt-edit mywindomain 'c:\autoexec.bat'

You can also edit files non-interactively (see L</NON-INTERACTIVE EDITING>
below).  To change the init default level to 5:

 virt-edit mydomain /etc/inittab -e 's/^id:.*/id:5:initdefault:/'

=head1 OPTIONS

=over 4

=item B<--help>

Display brief help.

=item B<--version>

Display version number and exit.

=item B<--backup extension> | B<-b extension>

Create a backup of the original file I<in the guest disk image>.  The backup
has the original filename with C<extension> added.

Usually the first character of C<extension> would be a dot C<.> so you would
write:

 virt-edit -b .orig [etc]

By default, no backup file is made.

=item B<--connect URI> | B<-c URI>

If using libvirt, connect to the given I<URI>.  If omitted, then we connect
to the default libvirt hypervisor.

If you specify guest block devices directly, then libvirt is not used at
all.

=item B<--format> raw

Specify the format of disk images given on the command line.  If this is
omitted then the format is autodetected from the content of the disk image.

If disk images are requested from libvirt, then this program asks libvirt
for this information.  In this case, the value of the format parameter is
ignored.

If working with untrusted raw-format guest disk images, you should ensure
the format is always specified.

=item B<--expr EXPR> | B<-e EXPR>

Instead of launching the external editor, non-interactively apply the Perl
expression C<EXPR> to each line in the file.  See L</NON-INTERACTIVE
EDITING> below.

Be careful to properly quote the expression to prevent it from being altered
by the shell.

=back

=head1 NON-INTERACTIVE EDITING

C<virt-edit> normally calls out to C<$EDITOR> (or vi) so the system
administrator can interactively edit the file.

There are two ways also to use C<virt-edit> from scripts in order to make
automated edits to files.  (Note that although you I<can> use C<virt-edit>
like this, it's less error-prone to write scripts directly using the
libguestfs API and Augeas for configuration file editing.)

The first method is to temporarily set C<$EDITOR> to any script or program
you want to run.  The script is invoked as C<$EDITOR tmpfile> and it should
update C<tmpfile> in place however it likes.

The second method is to use the I<-e> parameter of C<virt-edit> to run a
short Perl snippet in the style of L<sed(1)>.  For example to replace all
instances of C<foo> with C<bar> in a file:

 virt-edit domname filename -e 's/foo/bar/'

The full power of Perl regular expressions can be used (see L<perlre(1)>).
For example to delete root's password you could do:

 virt-edit domname /etc/passwd -e 's/^root:.*?:/root::/'

What really happens is that the snippet is evaluated as a Perl expression
for each line of the file.  The line, including the final C<\n>, is passed
in C<$_> and the expression should update C<$_> or leave it unchanged.

To delete a line, set C<$_> to the empty string.  For example, to delete the
C<apache> user account from the password file you can do:

 virt-edit mydomain /etc/passwd -e '$_ = "" if /^apache:/'

To insert a line, prepend or append it to C<$_>.  However appending lines to
the end of the file is rather difficult this way since there is no concept
of "last line of the file" - your expression just doesn't get called again.
You might want to use the first method (setting C<$EDITOR>) if you want to
do this.

The variable C<$lineno> contains the current line number.  As is
traditional, the first line in the file is number C<1>.

The return value from the expression is ignored, but the expression may call
C<die> in order to abort the whole program, leaving the original file
untouched.

Remember when matching the end of a line that C<$_> may contain the final
C<\n>, or (for DOS files) C<\r\n>, or if the file does not end with a
newline then neither of these.  Thus to match or substitute some text at the
end of a line, use this regular expression:

 /some text(\r?\n)?$/

Alternately, use the perl C<chomp> function, being careful not to chomp
C<$_> itself (since that would remove all newlines from the file):

 my $m = $_; chomp $m; $m =~ /some text$/

=head1 WINDOWS PATHS

C<virt-edit> has a limited ability to understand Windows drive letters and
paths (eg. C<E:\foo\bar.txt>).

If and only if the guest is running Windows then:

=over 4

=item *

Drive letter prefixes like C<C:> are resolved against the Windows Registry
to the correct filesystem.

=item *

Any backslash (C<\>) characters in the path are replaced with forward
slashes so that libguestfs can process it.

=item *

The path is resolved case insensitively to locate the file that should be
edited.

=back

There are some known shortcomings:

=over 4

=item *

Some NTFS symbolic links may not be followed correctly.

=item *

NTFS junction points that cross filesystems are not followed.

=back

=head1 USING GUESTFISH

L<guestfish(1)> is a more powerful, lower level tool which you can use when
C<virt-edit> doesn't work.

Using C<virt-edit> is approximately equivalent to doing:

 guestfish --rw -i -d domname edit /file

where C<domname> is the name of the libvirt guest, and C</file> is the full
path to the file.

The command above uses libguestfs's guest inspection feature and so does not
work on guests that libguestfs cannot inspect, or on things like arbitrary
disk images that don't contain guests.  To edit a file on a disk image
directly, use:

 guestfish --rw -a disk.img -m /dev/sda1 edit /file

where C<disk.img> is the disk image, C</dev/sda1> is the filesystem within
the disk image to edit, and C</file> is the full path to the file.

C<virt-edit> cannot create new files.  Use the guestfish commands C<touch>,
C<write> or C<upload> instead:

 guestfish --rw -i -d domname touch /newfile

 guestfish --rw -i -d domname write /newfile "new content"

 guestfish --rw -i -d domname upload localfile /newfile

C<virt-edit> cannot edit multiple files, but guestfish can do it like this:

 guestfish --rw -i -d domname edit /file1 : edit /file2

=head1 ЗМІННІ СЕРЕДОВИЩА

=over 4

=item C<EDITOR>

If set, this string is used as the editor.  It may contain arguments,
eg. C<"emacs -nw">

If not set, C<vi> is used.

=back

=head1 SHELL QUOTING

Libvirt guest names can contain arbitrary characters, some of which have
meaning to the shell such as C<#> and space.  You may need to quote or
escape these characters on the command line.  See the shell manual page
L<sh(1)> for details.

=head1 ТАКОЖ ПЕРЕГЛЯНЬТЕ

L<guestfs(3)>, L<guestfish(1)>, L<virt-cat(1)>, L<virt-copy-in(1)>,
L<virt-tar-in(1)>, L<Sys::Guestfs(3)>, L<Sys::Guestfs::Lib(3)>,
L<Sys::Virt(3)>, L<http://libguestfs.org/>, L<perl(1)>, L<perlre(1)>.

=head1 AUTHOR

Richard W.M. Jones L<http://people.redhat.com/~rjones/>

=head1 АВТОРСЬКІ ПРАВА

Copyright (C) 2009-2011 Red Hat Inc.

This program is free software; you can redistribute it and/or modify it
under the terms of the GNU General Public License as published by the Free
Software Foundation; either version 2 of the License, or (at your option)
any later version.

This program is distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
more details.

You should have received a copy of the GNU General Public License along with
this program; if not, write to the Free Software Foundation, Inc., 675 Mass
Ave, Cambridge, MA 02139, USA.

